### DISCUSSION TITLE
nio/nio-better-io

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#260] [Announcements]
# nio/nio-better-io

本篇内容主要讲述了 NIO 和传统 IO 之间的差异。首先，传统 IO 采用阻塞模型，而 NIO 使用非阻塞模型，通过选择器监控多个通道上的 I/O 事件，从而提升性能与可伸缩性。其次，在文件操作方面，传统 IO 依赖字节流或字符流进行文件读写，而 NIO 则利用通道和缓冲区进行操作，性能优势相对较小。最后，在网络传输方面，传统 IO 使用 Socket 和 ServerSocket，而 NIO 提供了 SocketChannel 和 ServerSocketChannel，支持非阻塞网络传输，进一步增强并发处理能力。

https://javabetter.cn/nio/nio-better-io.html

**Comment by Knightzh** [comment]
二哥，为什么我的测试时间是相反呢，我的IO是1297ms，NIO是4487ms，望二哥解答疑惑

**Reply by Crazylychee** [reply]
我也是

**Reply by zxbdzh** [reply]
+1

**Comment by Hanserwei** [comment]
测试结果不对，和二哥的相反。甚至前面拷贝文件的我都和二哥不一样。我测试1GB文件的时候，NIO还是会比传统IO快，但是如果用它的零拷贝，结果就是NIO完全碾压传统IO。至于后面的网络部分，测试流好多次，还是不行。结果出不来。
